anitya/authentication.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+
+"""
+Helper module for configuring OpenID Connect based authentication
+"""
+from functools import wraps
+import json
+
+import flask
+from flask_openid import OpenID
+from flask_oidc import OpenIDConnect
+
+APP = None
+
+
+####################################
+# Set up core OpenID Connect support
+####################################
+def configure_openid(app):
+    """Set up OpenID, OpenIDConnect, and the module's Flask app reference"""
+    global APP
+    app.oid = OpenID(app)
+    try:
+        app.oidc = OpenIDConnect(app, credentials_store=flask.session)
+    except Exception as exc:
+        # Handle running with only anonymous API access enabled
+        app.logger.debug(str(exc))
+        app.oidc = None
+    app.route("/oidc_callback")(register_oidc_client)
+    APP = app
+
+
+def register_oidc_client(code, state):
```

Comment:
I'd prefer to leave fleshing that out until the TODO comment is addressed by https://github.com/release-monitoring/anitya/issues/442